api_bonus.yaml

```
@@ -0,0 +1,144 @@
+swagger: "2.0"
+info:
+  version: 1.0.0
+  title: Bonus API
+host: localhost
+basePath: /api/v1
+schemes:
+  - https
+security:
+  - Bearer: []
+paths:
+  "/bonus/vacanze/request":
+    post:
+      operationId: startBonusEligibilityCheck
+      summary: Start bonus eligibility check (ISEE)
+      responses:
+        '202':
+          description: Request accepted.
+          schema:
+            type: object
+            properties:
+              id:
+                type: string
+                description: The identifier of the created request.
+          headers:
+            Location:
+              type: string
+              description: |-
+                Location (URL) of created request resource.
+                A GET request to this URL returns the request status and details.
+          examples:
+            application/json:
+              id: 01BX9NSMKVXXS5PSP2FATZMYYY
+
+    get:
+      operationId: getBonusEligibilityCheck
+      summary: Get eligibility (ISEE) check information for user's bonus
+      responses:
+        '202':
+          description: Processing request.
+          schema:
+            type: object
+            properties:
+              id:
+                type: string
+                description: The identifier of the created request.
+          examples:
+            application/json:
+              id: 01BX9NSMKVXXS5PSP2FATZMYYY
+        '200':
+          description: Request processed.
+          schema:
+            type: object
+            properties:
+              id:
+                type: string
+                description: The identifier of the created request.
+          examples:
+            application/json:
+              # REQUEST (isee): NO_DATA, ELIGIBLE, INELIGIBLE, NOT_FOUND
+              status: ELIGIBLE
+              members:
+                - name: "Mario"
+                  surname: "Rossi"
+                - name: "Piero"
+                  surname: "Rossi"
+                - name: "Giulia"
+                  surname: "Rossi"
+              max_amount: 50000
+              tax_benefit: 3000
+
+    delete:
+      operationId: cancelBonusEligibilityProcedure
+      summary: Cancel bonus eligibility check procedure (avoids sending a push notification when done)
+      responses:
+        '200':
+          description: Request canceled.
+          schema:
+            type: object
+            properties:
+              id:
+                type: string
+                description: The identifier of the canceled request.
+          examples:
+            application/json:
+              id: 01BX9NSMKVXXS5PSP2FATZMYYY
+
+  "/bonus/vacanze":
```

for consistency with the `eligibility` endpoint, I would make it explicit that this is an activation request, e.g.

```suggestion
  "/bonus/vacanze/activate":
```

---

we should also have an `/bonus/vacanze/active` that returns the active bonus if there's one

---

why do you want two different endpoints for the same result ? there's only one bonus for each user and it can be in any state (the entity is versioned)